django/db/models/fields/tuple_lookups.py

```diff
@@ -124,6 +125,20 @@ def get_fallback_sql(self, compiler, connection):
         )
 
     def as_sql(self, compiler, connection):
+        if (
```

Comment:
Oracle <19 was failing because the mixin I added with this logic was not ordered correctly with this logic. Rather than have two mixins, which seems a little fragile, I just brought new logic here, and guarded with an `isinstance` check. Seems like a pick your thorn thing. 🤷‍♂️